### PR TITLE
DROOLS-2304: Cover reaction on ExpressionEditorSelectedEvent

### PR DESCRIPTION
**Pull request by jomarko**
Dependent on #1415 

Few more tests and small change by `Objects.equals` nazi :)

@manstis please have a look.

I will rebase once your PR is in.

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package org.kie.workbench.common.dmn.client.editors.expressions;
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorTest.java]: License police are equally vigilant!

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        verify(view).onExpressionEditorSelected(expressionGrid);
+    }
+
+    @Test
+    public void testObservesExpressionEditorSelectedEvent_DifferentSessions() throws Exception {
+        final ExpressionEditorSelectedEvent firedEvent = mock(ExpressionEditorSelectedEvent.class);
+        final Optional<BaseExpressionGrid> expressionGrid = Optional.of(mock(BaseExpressionGrid.class));
+
+        doReturn(expressionGrid).when(firedEvent).getEditor();
+        // same client sessions
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorTest.java]: Actually they're different client sessions!

**Comment by manstis** [comment]
@jomarko You can rebase this now.

**Comment by jomarko** [comment]
@manstis License added and test comment updated

**Comment by jomarko** [comment]
@manstis just friendly reminder, tests passed.

**Pull request merged**